src/main/java/org/folio/dataexp/service/export/strategies/HoldingsExportAllStrategy.java

```diff
@@ -23,112 +21,131 @@
+  private Slice<HoldingsRecordEntity> nextFolioSlice(JobExecutionExportFilesEntity exportFilesEntity, ExportRequest exportRequest, Pageable pageble) {
     if (Boolean.TRUE.equals(exportRequest.getSuppressedFromDiscovery())) {
-      return holdingsRecordEntityRepository.findByIdGreaterThanEqualAndIdLessThanEqualOrderByIdAsc(exportFilesEntity.getFromId(),
-        exportFilesEntity.getToId(), pageble);
+      return folioHoldingsAllRepository.findFolioHoldingsAllNonDeleted(exportFilesEntity.getFromId(), exportFilesEntity.getToId(),
+          pageble);
     }
-    return holdingsRecordEntityRepository.findAllWhenSkipDiscoverySuppressed(exportFilesEntity.getFromId(),
-      exportFilesEntity.getToId(), pageble);
+    return folioHoldingsAllRepository.findFolioHoldingsAllNonDeletedNonSuppressed(exportFilesEntity.getFromId(),
+        exportFilesEntity.getToId(), pageble);
   }
 
-  private List<HoldingsRecordEntity> holdingsDeletedToHoldingsEntities(List<HoldingsRecordDeletedEntity> holdingsDeleted) {
-    return holdingsDeleted.stream()
-      .map(hold -> new HoldingsRecordEntity().withId(UUID.fromString(getAsJsonObject(getAsJsonObject(hold.getJsonb()).get()
-          .getAsString(DELETED_AUDIT_RECORD)).get()
-          .getAsString("id")))
-        .withJsonb(getAsJsonObject(hold.getJsonb()).get()
-          .getAsString(DELETED_AUDIT_RECORD))
-        .withInstanceId(UUID.fromString(getAsJsonObject(getAsJsonObject(hold.getJsonb()).get()
-          .getAsString(DELETED_AUDIT_RECORD)).get()
-          .getAsString("instanceId"))))
-      .toList();
+  private Slice<MarcRecordEntity> nextMarcSlice(JobExecutionExportFilesEntity exportFilesEntity, ExportRequest exportRequest, Pageable pageble) {
+    if (Boolean.TRUE.equals(exportRequest.getSuppressedFromDiscovery())) {
+      return marcHoldingsAllRepository.findMarcHoldingsAllNonDeleted(exportFilesEntity.getFromId(), exportFilesEntity.getToId(),
+          pageble);
+    }
+    return marcHoldingsAllRepository.findMarcHoldingsAllNonDeletedNonSuppressed(exportFilesEntity.getFromId(), exportFilesEntity.getToId(),
+        pageble);
   }
 
-  private void createAndSaveMarc(Set<UUID> holdingsIds, List<HoldingsRecordEntity> holdings, ExportStrategyStatistic exportStatistic,
-      MappingProfile mappingProfile, UUID jobExecutionId, ExportRequest exportRequest) {
-    var externalIdsWithMarcRecord = new HashSet<UUID>();
-    createMarc(holdingsIds, exportStatistic, mappingProfile, jobExecutionId, exportRequest, externalIdsWithMarcRecord);
-    holdings.removeIf(hold -> externalIdsWithMarcRecord.contains(hold.getId()));
-    var result = getGeneratedMarc(holdingsIds, holdings, mappingProfile, jobExecutionId, exportRequest);
-    saveMarc(result, exportStatistic);
+  private List<HoldingsRecordEntity> getFolioDeleted(ExportRequest exportRequest) {
+    if (Boolean.TRUE.equals(exportRequest.getSuppressedFromDiscovery())) {
```

Comment:
@obozhko-folio 
I believe the boolean values equality check is not needed.
(exportRequest.getSuppressedFromDiscovery()) - is enough for the condition check.

Reply:
if you believe, such a case should be improved, check for the same code block entries through all code changes.
Thanks

Reply:
What if exportRequest.getSuppressedFromDiscovery() is null?